sbt_common/messaging/src/main/scala/uk/ac/wellcome/messaging/sns/SNSMessageWriter.scala

```diff
@@ -0,0 +1,45 @@
+package uk.ac.wellcome.messaging.sns
+
+import com.amazonaws.services.sns.AmazonSNS
+import com.amazonaws.services.sns.model.PublishRequest
+import com.google.inject.Inject
+import grizzled.slf4j.Logging
+import io.circe.Encoder
+import uk.ac.wellcome.json.JsonUtil._
+
+import scala.concurrent.{blocking, ExecutionContext, Future}
+
+/** Writes messages to SNS.  This class can write to a different topic on
+  * every request.
+  *
+  */
+class SNSMessageWriter @Inject()(snsClient: AmazonSNS)(
+  implicit ec: ExecutionContext)
+    extends Logging {
+  def writeMessage(message: String,
```

Comment:
Might be worth allowing currying here, i.e.

```scala
def writeMessage(topicArn: String)(message: String, subject: String) = {
  // ...
}
```

Then you can create a writer like:

```scala
val writer = writer.writeMessage(topicArn) _

writer("subject", "message")
writer("anotherSubject", "message")
```